examples/hotrod/services/frontend/server.go

```diff
@@ -72,16 +73,17 @@ func NewServer(options ConfigOptions, tracer opentracing.Tracer, logger log.Fact
 func (s *Server) Run() error {
 	mux := s.createServeMux()
 	s.logger.Bg().Info("Starting", zap.String("address", "http://"+path.Join(s.hostPort, s.basepath)))
-	return http.ListenAndServe(s.hostPort, mux)
+	return http.ListenAndServe(s.hostPort, otelhttp.NewHandler(mux, s.basepath,
+		otelhttp.WithTracerProvider(s.tracer)))
 }
 
 func (s *Server) createServeMux() http.Handler {
-	mux := tracing.NewServeMux(true, s.tracer, s.logger)
+	// mux := tracing.NewServeMux(true, s.tracer, s.logger)
 	p := path.Join("/", s.basepath)
-	mux.Handle(p, http.StripPrefix(p, http.FileServer(s.assetFS)))
-	mux.Handle(path.Join(p, "/dispatch"), http.HandlerFunc(s.dispatch))
-	mux.Handle(path.Join(p, "/config"), http.HandlerFunc(s.config))
-	return mux
+	s.mux.Handle(p, otelhttp.WithRouteTag(p, http.StripPrefix(p, http.FileServer(s.assetFS))))
```

Comment:
s.mux is never initialized. Why does it need to be stored?

Reply:
This function is asked to create & return a handler, which happens to be implemented as Mux. Why does it need to be stored in `Server` instead of created locally here, as was done before?